src/mds/MDBalancer.cc

```diff
@@ -1404,11 +1404,17 @@ void MDBalancer::handle_mds_failure(mds_rank_t who)
   }
 }
 
-int MDBalancer::dump_loads(Formatter *f) const
+int MDBalancer::dump_loads(Formatter *f, int64_t depth) const
 {
-  std::deque<CDir*> dfs;
+  std::deque<pair<CDir*, int>> dfs;
+  std::deque<CDir*> dfs_root;
```

Comment:
Seems like we are just concerned with `front()` of the deque. We peek at the front, pop the element from there and push it to another deque `dfs`. Maybe we use stack here? The rationale is we just need one pointer for our operation (peek and pop from top). Thoughts?

Reply:
Did you mean deque dfs_root? It's just a temporary place for root dir frags since I don't want to change the output of get_dirfrags function (called elsewhere) to include depth info. Ths main logic is about dfs deque afterwards.
Besides, change deque dfs_root to stack may also need to change  function get_dirfrags.

Reply:
Okay. Looks good.